File: backend/mcp_server/tools/cases/parties.py
"""案件当事人 MCP tools"""

from __future__ import annotations

from typing import Any

from mcp_server.client import client


def list_case_parties(case_id: int) -> list[dict[str, Any]]:
    """查询指定案件的所有当事人，包含客户详情和诉讼地位。"""
    return client.get("/cases/parties", params={"case_id": case_id})  # type: ignore[return-value]


def add_case_party(case_id: int, client_id: int, legal_status: str | None = None) -> dict[str, Any]:
    """为案件添加当事人。legal_status 为诉讼地位，如：原告、被告、第三人等。"""
    payload: dict[str, Any] = {"case_id": case_id, "client_id": client_id}
    if legal_status:
        payload["legal_status"] = legal_status
    return client.post("/cases/parties", json=payload)  # type: ignore[return-value]

File: frontend/src/features/cases/api/parties.ts
import { createFeatureApiClient } from '@/lib/api'
import type { CaseParty, CaseAssignment, CaseAccessGrant } from '../types'

const client = createFeatureApiClient('cases')

export const partiesApi = {
  list: async (caseId: number | string): Promise<CaseParty[]> =>
    client.get('parties', { searchParams: { case_id: String(caseId) } }).json<CaseParty[]>(),

  create: async (data: { case_id: number; client_id: number; legal_status?: string }): Promise<CaseParty> =>
    client.post('parties', { json: data }).json<CaseParty>(),

  update: async (id: number | string, data: { case_id?: number; client_id?: number; legal_status?: string }): Promise<CaseParty> =>
    client.put(`parties/${id}`, { json: data }).json<CaseParty>(),

  delete: async (id: number | string): Promise<void> => {
    await client.delete(`parties/${id}`)
  },

  listAssignments: async (caseId: number | string): Promise<CaseAssignment[]> =>
    client.get('assignments', { searchParams: { case_id: String(caseId) } }).json<CaseAssignment[]>(),

  createAssignment: async (data: { case_id: number; lawyer_id: number }): Promise<CaseAssignment> =>
    client.post('assignments', { json: data }).json<CaseAssignment>(),

  deleteAssignment: async (id: number | string): Promise<void> => {
    await client.delete(`assignments/${id}`)
  },

  listGrants: async (caseId: number | string): Promise<CaseAccessGrant[]> =>
    client.get('grants', { searchParams: { case_id: String(caseId) } }).json<CaseAccessGrant[]>(),

  createGrant: async (data: { case_id: number; grantee_id: number }): Promise<CaseAccessGrant> =>
    client.post('grants', { json: data }).json<CaseAccessGrant>(),

  deleteGrant: async (id: number | string): Promise<void> => {
    await client.delete(`grants/${id}`)
  },
}
